dashboard/src/reducers/tableOfContentReducer.js

```diff
@@ -52,10 +53,9 @@ const TableOfContentReducer = (state = initialState, action = {}) => {
       };
 
     case UPDATE_STACK:
-      return {
-        ...state,
-        stack: payload,
-      };
+      const newState = { ...state };
+      newState.stack.length = payload;
+      return newState;
```

Comment:
By the way, my apologies for the bad lead in [my previous comment](https://github.com/distributed-system-analysis/pbench/pull/2952#discussion_r929175756), I should have known better....

Reply:
I am highly afraid this is also not the correct step.The code provided above does not change the length of stack, it just appends a length key to the object ,and only the length key changes.Now suppose I clicked on drop down and went back to a previous subdirectory, the stack is not slimmed down ,but it is still that bulkier stack with all the subdirectories we have visited upto now and if I now click on the subdirectory to go to the other one, ...state.stack has become uniterable.

Am I missing something?

I looked for ways to make deep copy of an object ,and found some solutions.I will attach some links here, since I don't know which one we can follow or even if we want to follow these approach or not.

[Using structured clone](https://web.dev/structured-clone/) [I have some questions regarding browser support for this approach]
[Some other possible ways](https://code.tutsplus.com/articles/the-best-way-to-deep-copy-an-object-in-javascript--cms-39655)

Reply:
> I am highly afraid this is also not the correct step.

Your instincts are excellent!

> By the way, my apologies for the bad lead in https://github.com/distributed-system-analysis/pbench/pull/2952#discussion_r929175756, I should have known better....

I **_did_** know better!  I had forgotten that `.length` is not a field in an object, because `stack` is an array, not an object!  (I'm still reeling from the idea that Javascript lets you write to the array length in a way which truncates the array....)

> I looked for ways to make deep copy of an object

Based on what I've read, it's not obvious that we actually want a deep copy.  What we want to be sure of is that we produce a new (top-level) object whose references to subobjects point to new objects for subobjects which have been mutated.  That is, when the JS runtime (or Redux, or React, or whoever) walks the new state comparing it to the old state, it should never find any fields are equal when those refer to objects which have been modified in some way.  So, just as we must not mutate the top-level object, we must not mutate any nested objects -- we have to replace them with new objects -- however, for nested objects (and, by "object" here, I'm including arrays...) which are unchanged, I see no reason not to reuse them in the new top-level object.

So, rather than worrying about how to produce a deep copy (which might hard, or resource-intensive, or slow), I would instead concentrate on producing a new object which is a shallow copy, but with an additional shallow copy at each level until we get down to the actual change (and I think the "spread" operator is quite adequate for the purpose).

But, anyway, yeah, I blew it with `length`.  I suspect that the proper approach there is to use `slice()` instead -- it returns a shallow copy of the array.  So, something like this:
```
    case UPDATE_STACK:
      return {
        ...state,
        stack: state.stack.slice(0, payload)
        }
      }
```